contracts/wizards/v1/listing/ListingWizardV1.sol

```diff
@@ -65,4 +187,57 @@ contract ListingWizardV1 is IListingWizardV1, ContractEntity, Multicall {
     function supportsInterface(bytes4 interfaceId) public view override(ContractEntity, IERC165) returns (bool) {
         return interfaceId == type(IListingWizardV1).interfaceId || super.supportsInterface(interfaceId);
     }
+
+    /**
+     * @dev "Consume a nonce": return the current value and increment.
+     *
+     * _Available since v4.1._
+     */
+    function _useNonce(address lister) internal returns (uint256 current) {
+        CountersUpgradeable.Counter storage delegatedListingNonces = _listerDelegatedListingNonces[lister];
+        current = delegatedListingNonces.current();
+        delegatedListingNonces.increment();
+    }
+
+    function _createListingWithTerms(
+        Assets.Asset[] calldata assets,
+        Listings.Params calldata params,
+        IListingTermsRegistry.ListingTerms calldata terms,
+        uint32 maxLockPeriod,
+        bool immediatePayout,
+        uint256 universeId
+    ) internal returns (uint256 listingId, uint256 listingTermsId) {
+        listingId = IListingManager(_metahub.getContract(Contracts.LISTING_MANAGER)).createListing(
+            assets,
+            params,
+            maxLockPeriod,
+            immediatePayout
+        );
+
+        GeneralWizardsHelperV1.checkListingTermsAreValid(terms);
+        listingTermsId = IListingTermsRegistry(_metahub.getContract(Contracts.LISTING_TERMS_REGISTRY))
+            .registerUniverseListingTerms(listingId, universeId, terms);
+
+        // Detecting address of Original Assets Collection (all Assets are from the same Collection)
+        address originalCollection = assets[0].token();
+
+        ListingWizardsHelperV1.validateMatchWithUniverse(universeId, originalCollection, terms, _metahub);
+    }
+
+    function _validateSigner(
+        address declaredLister,
```

Comment:
in _validateDelegatedBulkListingSigner:

declaredLister -> declaredBulkLister
delegatedListingHash -> delegatedBulkListingHash
delegatedListingSignature -> delegatedBulkListingSignature